fdcache/fdcache.go

```diff
@@ -155,3 +159,17 @@ func GetAndForgetFile(conn net.Conn) *os.File {
 	delete(cache, key)
 	return entry.Fp // Pass ownership to caller
 }
+
+// GetUUID returns the UUID for a passed-in connection.
+func GetUUID(conn net.Conn) (string, error) {
+	key := makekey(conn)
+	mutex.Lock()
+	defer mutex.Unlock()
+	entry, found := cache[key]
+	if !found {
+		return "", fmt.Errorf("fd not found")
+	}
+	id, err := uuid.FromFile(entry.Fp)
+	log.Println("UUID:", id)
```

Comment:
This should be `log.Debugf("UUID: %s", id)` where `log` is `apex/log` (or, if it is just meant as a debugging aid, it should be removed before merging).